suite/tests/client-interface/avx512ctx.c

```diff
@@ -0,0 +1,306 @@
+ *   specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL GOOGLE, INC. OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ */
+
+#ifndef ASM_CODE_ONLY /* C code */
+#    include "tools.h"
+
+#    define VERBOSE 0
+
+NOINLINE void
+marker();
+/* asm routines */
+void
+init_zmm(byte *buf);
+void
+get_zmm(byte *buf);
+void
+init_opmask(byte *buf);
+void
+get_opmask(byte *buf);
+
+void
+run_avx512()
+{
+#    ifndef __AVX512F__
+#        error "Build error, should only be added with AVX-512 support."
+#    endif
+#    ifdef X64
+#        define NUM_SIMD_REGS 32
+#    else
+#        define NUM_SIMD_REGS 8
+#    endif
+#    define NUM_OPMASK_REGS 8
+    byte zmm_buf[NUM_SIMD_REGS * 64];
+    byte zmm_ref[NUM_SIMD_REGS * 64];
+    byte opmask_buf[NUM_OPMASK_REGS * 2];
+    byte opmask_ref[NUM_OPMASK_REGS * 2];
+    memset(zmm_buf, 0xde, sizeof(zmm_buf));
+    memset(zmm_ref, 0xab, sizeof(zmm_ref));
+    /* Even though DynamoRIO is capable of handling AVX512BW wide 64-bit mask registers,
+     * we're simplifying the test here and are checking only AVX512F wide 16-bit mask
+     * registers. This also lets us run the test in 32-bit mode.
+     */
+    memset(opmask_buf, 0xde, sizeof(opmask_buf));
+    memset(opmask_ref, 0xab, sizeof(opmask_ref));
+
+    init_zmm(zmm_ref);
+    init_opmask(opmask_ref);
+    marker();
+    get_zmm(zmm_buf);
+    get_opmask(opmask_buf);
+
+#    if VERBOSE
+    int zmm_off = 0;
+    for (int i = 0; i < NUM_SIMD_REGS; ++i, zmm_off += 64) {
+        print("got zmm[%d]:\n", i);
+        for (int b = zmm_off; b < zmm_off + 64; ++b)
+            print("%x", zmm_buf[b]);
+        print("\nref zmm[%d]:\n", i);
+        for (int b = zmm_off; b < zmm_off + 64; ++b)
+            print("%x", zmm_ref[b]);
+        print("\n");
+    }
+    int opmask_off = 0;
+    for (int i = 0; i < NUM_OPMASK_REGS; ++i, opmask_off += 2) {
+        print("got k[%i]:\n", i);
+        for (int b = opmask_off; b < opmask_off + 2; ++b)
+            print("%x", opmask_buf[b]);
+        print("\nref k[%d]:\n", i);
+        for (int b = opmask_off; b < opmask_off + 2; ++b)
+            print("%x", opmask_ref[b]);
+        print("\n");
+    }
+#    endif
+
+    if (memcmp(zmm_buf, zmm_ref, sizeof(zmm_buf)))
```

Comment:
style: use booleans for booleans, not integers

Reply:
Fixed.